02-microarray/pathway-analysis_microarray_03_gsva.Rmd

```diff
@@ -405,11 +424,222 @@ Note that if we had duplicate gene identifiers here, we would not be able to set
+
+In this contrasts matrix, we are comparing each subgroup to the average of the other subgroups.  
+We're dividing by two in this expression so that each group is compared to the average of the other two groups (`makeContrasts()` doesn't allow you to use functions like `mean()`; it wants a formula).
+
+```{r}
+contrast_matrix <- makeContrasts(
+  "G3vsOther" = Group3 - (Group4 + SHH) / 2,
+  "G4vsOther" = Group4 - (Group3 + SHH) / 2,
+  "SHHvsOther" = SHH - (Group3 + Group4) / 2,
+  levels = des_mat
+)
+```
+
+Side note: If you did have a control group you wanted to compare each group to, you could make each contrast to that control group, so the formulate would look like `Group3 = Group3 - Control` for each one.
+We highly recommend consulting the [limma users guide](https://bioconductor.org/packages/release/bioc/vignettes/limma/inst/doc/usersguide.pdf) for figuring out what your `makeContrasts()` and `model.matrix()` setups should look like [@Ritchie2015].
+
+Now that we have the contrasts matrix set up, we can use it to re-fit the model and re-smooth it with `eBayes()`.
+
+```{r}
+# Fit the model according to the contrasts matrix
+contrasts_fit <- contrasts.fit(fit, contrast_matrix)
+
+# Re-smooth the Bayes
+contrasts_fit <- eBayes(contrasts_fit)
+```
+
+Here's a [nifty article and example](http://varianceexplained.org/r/empirical_bayes_baseball/) about what the empirical Bayes smoothing is for [@bayes-estimates].
+
+Now let's create the results table based on the contrasts fitted model.
+
+This step will provide the Benjamini-Hochberg multiple testing correction.
+The `topTable()` function default is to use Benjamini-Hochberg but this can be changed to a different method using the `adjust.method` argument (see the `?topTable` help page for more about the options).
+
+```{r}
+# Apply multiple testing correction and obtain stats
+stats_df <- topTable(contrasts_fit, number = nrow(expression_df)) %>%
+  tibble::rownames_to_column("Gene")
+```
+
+Let's take a peek at our results table.
+
+```{r}
+head(stats_df)
+```
+
+For each gene, each group's fold change in expression, compared to the average of the other groups is reported.
+
+By default, results are ordered from largest `F` value to the smallest, which means your most differentially expressed genes across all groups should be toward the top.
+
+This means `HALLMARK_UNFOLDED_PROTEIN_RESPONSE` appears to be the pathway that contains the most significant distribution of scores across subgroups. 
+
+## Visualizing Results
+
+Let's make a plot for our most significant pathway, `HALLMARK_UNFOLDED_PROTEIN_RESPONSE`.
+
+### Sina plot
+
+First we need to get our GSVA scores for this pathway in an appropriate format for `ggplot2`.
+
+```{r}
+annotated_results_df <-
+  # Pull out only the results for this pathway
+  gsva_results["HALLMARK_UNFOLDED_PROTEIN_RESPONSE", ] %>%
```

Comment:
I think a _more general_ approach where we are getting the `gsva_results` matrix (an assumption, could be a wide data frame) into a long format could be welcome here. I suspect we don't do a lot of wide -> long in these examples and it is something that comes up pretty often when you go to plot these types of data. This has come up before on the draft PR https://github.com/AlexsLemonade/refinebio-examples/pull/352#discussion_r524295683. (To be fair, I'm the one that brought it up each time.)

I anticipate that we'd want to break this up into two or three steps instead of doing it "all at once":

- Make long format
- Maybe filter to relevant pathway 
- Join metadata

(Filter before join makes sense to me generally but depends on how we're going to write about it and what we want to show!)